packages/flutter_tools/lib/src/macos/application_package.dart

```diff
@@ -58,8 +63,26 @@ abstract class MacOSApp extends ApplicationPackage {
       }
       bundleDir = globals.fs.directory(applicationBundle);
     } else {
-      globals.printError('Folder "${applicationBundle.path}" is not an app bundle.');
-      return null;
+      // Try to unpack as a zip.
+      final Directory tempDir = globals.fs.systemTempDirectory.createTempSync('flutter_app.');
+      shutdownHooks.addShutdownHook(() async {
```

Comment:
I don't believe this is necessary because the tool should delete an outstanding directories during shutdown. (unless it receives a sigkill, but then the shutdown hook won't run anyway)

Reply:
I see that now in systemTempDirectory, cool.